src/modules/position_estimator_inav/position_estimator_inav_main.cpp

```diff
@@ -773,67 +782,124 @@ int position_estimator_inav_thread_main(int argc, char *argv[])
 
 			/* check no vision circuit breaker is set */
 			if (params.no_vision != CBRK_NO_VISION_KEY) {
-				/* vehicle vision position */
-				orb_check(vision_position_sub, &updated);
+				/* vehicle visual odometry */
+				orb_check(visual_odom_sub, &updated);
 
 				if (updated) {
-					orb_copy(ORB_ID(vehicle_vision_position), vision_position_sub, &vision);
+					orb_copy(ORB_ID(vehicle_visual_odometry), visual_odom_sub, &visual_odom);
 
 					static float last_vision_x = 0.0f;
 					static float last_vision_y = 0.0f;
 					static float last_vision_z = 0.0f;
 
+					vision_xy_valid = !PX4_ISFINITE(visual_odom.pose_covariance[0]) ? sqrtf(fmaxf(visual_odom.pose_covariance[0],
```

Comment:
inverted logic

Reply:
yeah. The reason is that I was using the `PX4_ISNAN` on the previous PR and forget to change the logic. I will add the `PX4_ISNAN` after.